### PR TITLE
Adsk Contrib - Replace tinyXML by expat (#57)

### PR DESCRIPTION
**Pull request by BernardLefebvre**
eXpat is the xml parsing solution chosen for clf parsing, Replacing use of tinyXML by eXpat.
Add support for SATNode for CDL xml parsing.
This is a step towards CLF support https://github.com/imageworks/OpenColorIO/issues/610

### CODE REVIEW

File: src/OpenColorIO/fileformats/FileFormatIridasLook.cpp [hunk 1]
```diff
@@ -150,223 +151,417 @@ OCIO_NAMESPACE_ENTER
...
+                    what = pystring::replace(what, " ", "");
+                    what = pystring::replace(what, "\"", "");
+                    what = pystring::replace(what, "'", "");
+                    what = pystring::replace(what, "\n", "");
+                    // Append to lut string
+                    pImpl->m_lutString += what;
                 }
             }
 
+            unsigned getXmLineNumber() const
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/FileFormatIridasLook.cpp]:
```suggestion
            unsigned getXmlLineNumber() const
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,1060 @@
...
+
+void XmlReaderElementStack::clear()
+{
+    m_elms.clear();
+}
+
+std::string XmlReaderElementStack::dump() const
+{
+    std::ostringstream errMsg;
+    errMsg << "CTF stack is ";
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp]:
```suggestion
    errMsg << "CDL stack is ";
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,1060 @@
...
+        std::string value(original1);
+        OCIO::LTrim(value);
+        OIIO_CHECK_ASSERT(0 == strcmp("some text    ", value.c_str()));
+        value = original2;
+        OCIO::LTrim(value);
+        OIIO_CHECK_ASSERT(0 == strcmp("some text  \t \v \f ", value.c_str()));
+    }
+}
+
+OIIO_ADD_TEST(CTFReaderUtil, ParseNumber)
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp]:
```suggestion
OIIO_ADD_TEST(CDLReaderUtil, ParseNumber)
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp [hunk 4]
```diff
@@ -0,0 +1,1060 @@
...
+        std::string buffer("            1");
+        OIIO_CHECK_THROW_WHAT(OCIO::ParseNumber(buffer.c_str(),
+                                                buffer.size(),
+                                                data),
+                              OCIO::Exception,
+                              "not start with space");
+    }
+}
+
+OIIO_ADD_TEST(CTFReaderUtil, FindSubString)
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp]:
```suggestion
OIIO_ADD_TEST(CDLReaderUtil, FindSubString)
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h [hunk 5]
```diff
@@ -0,0 +1,771 @@
...
+    virtual bool isContainer() const = 0;
+
+    const std::string& getName() const
+    {
+        return m_name;
+    }
+
+    virtual const std::string& getIdentifier() const = 0;
+
+    unsigned getXmLineNumber() const
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]:
```suggestion
    unsigned getXmlLineNumber() const
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h [hunk 6]
```diff
@@ -0,0 +1,771 @@
...
+    void start(const char **atts)
+    {
+        m_isSlopeInit = m_isOffsetInit = m_isPowerInit = false;
+    }
+
+    void end()
+    {
+        if (!m_isSlopeInit)
+        {
+            throwMessage("CTF CDL parsing. Required node 'Slope' is missing. ");
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]:
```suggestion
            throwMessage("CDL parsing. Required node 'Slope' is missing. ");
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h [hunk 7]
```diff
@@ -0,0 +1,771 @@
...
+        }
+
+        if (!m_isOffsetInit)
+        {
+            throwMessage("CTF CDL parsing. Required node 'Offset' is missing. ");
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]:
```suggestion
            throwMessage("CDL parsing. Required node 'Offset' is missing. ");
```

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h [hunk 8]
```diff
@@ -0,0 +1,771 @@
...
+        }
+
+        if (!m_isPowerInit)
+        {
+            throwMessage("CTF CDL parsing. Required node 'Power' is missing. ");
```
Comment (michdolan) [inline src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]:
```suggestion
            throwMessage("CDL parsing. Required node 'Power' is missing. ");
```

File: src/OpenColorIO/fileformats/FileFormatIridasLook.cpp (hunk 1 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/FileFormatIridasLook.cpp]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp (hunk 2 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp (hunk 3 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.cpp]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h (hunk 5 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h (hunk 6 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h (hunk 7 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]: implemented by the next commit

File: src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h (hunk 8 above)
Comment (BernardLefebvre) [reply src/OpenColorIO/fileformats/cdl/CDLReaderHelper.h]: implemented by the next commit

**Comment by michdolan** [comment]
This PR is now 15 days old with one approval, which makes it acceptable to merge per our policy.

**Pull request merged**